lib/neovim-plugin.nix

```diff
@@ -113,6 +113,30 @@
                 '';
                 internal = true;
               };
+
+              lazyLoad = lib.mkOption {
+                default = {
+                  enable = false;
+                };
+                description = ''
+                  Lazy load configuration settings.
+                '';
+                type = lib.types.submodule {
+                  options = {
+                    enable = lib.mkOption {
+                      default = false;
+                      type = lib.types.bool;
+                      description = ''
+                        Whether to lazy load the plugin.
+
+                        If you enable this, the plugin's lua configuration will need to be manually loaded by other means.
+
+                        A usage would be passing the plugin's luaConfig to the `plugins.lz-n.plugins` configuration.
```

Comment:
Q: I like this initial implementation, but do you think it is too much of a stretch to go straight to implementing this ourselves in this PR?

That way we never have the interim stage of end-users having to enable a "dont setup" option and passing their plugin's settings to another plugin which lazy-loads them.

Reply:
I have another branch i'm playing around with right now for setting up lz-n for a user following the work done in https://github.com/nix-community/nixvim/pull/1875

Reply:
Yeah... I thought about that when I started going through and implementing my own lazy loading. It felt tedious enough, let alone if I had to redo it again.